### PR TITLE
Custom file handler, shutdown, close pipe

### PR DESCRIPTION
**Pull request by msnidal**
I added a custom file handler to save the log files to the user's home folder instead of where they happened to run the program - these are organized and formatted by date. Shutdown now uses the logger.shutdown() function, and the pipe is closed from the "inside" (ie after "done" is sent, shutdown and pipe close happen.)

I also updated the readme to use markdown, he he he

### CODE REVIEW

File: roboticslogger/file_handler.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import logging
+import os
+import errno
+import datetime
+
+def get_timestamped_filepath():
+    """ Gets a timestamped filepath for the logfiles """
+    return "{0}/Logs/RoboticsLogger/".format(os.path.expanduser('~'))
```
Comment (psyomn) [inline roboticslogger/file_handler.py]: To be safe, you might want to use something that the library provides to build fs filepaths. For example to achieve the above you'd do this in ruby:

``` ruby
path = File.join("path", "to", "file")
# "path/to/file"
```

By delegating that to some library, or the std library of whatever language you're using, you're able to not worry about cross platform problems. (A nice reading would be the rust [PathBuf](https://doc.rust-lang.org/std/path/struct.PathBuf.html)... please don't kill me).

Comment (psyomn) [inline roboticslogger/file_handler.py]: Also, maybe consider using something like this:

``` python
os.path.expanduser('~/.config/rover/logs')
```

File: roboticslogger/logger.py [hunk 2]
```diff
@@ -44,8 +46,10 @@ def run(self, conn):
             elif msg[0] == "crit":
                 self.crit(msg[1])
             else:
-                error("Cannot log properly msg.")
+                self.err("Failed to log message.")
```
Comment (psyomn) [inline roboticslogger/logger.py:49]: Can you log this? .... :stuck_out_tongue:

File: roboticslogger/file_handler.py (hunk 1 above)
Comment (msnidal) [inline roboticslogger/file_handler.py]: Interesting. I'm playing with it on Windows now, and it seems like python's os module actually provides this functionality under os.join. I'll try it again on my linux box then send a commit.

File: roboticslogger/logger.py (hunk 2 above)
Comment (msnidal) [inline roboticslogger/logger.py:49]: LOL this is the case where the pipe receives some garbage input (ie a list with an improperly named first element.)

**Pull request merged**